app/views/admin/AdministerUserModal.coffee

```diff
@@ -274,6 +277,16 @@ module.exports = class AdministerUserModal extends ModalView
       }
     }).then (res) =>
       @userSaveState = 'saved'
+      permissions = @user.get('permissions')
+
+      if checked and res.schoolAdministrator == 0
```

Comment:
Because this is an admin tool it probably doesn't need to be the most robust thing. But out of interest should you get the state of `checked` again after the initial promise resolves in case the user was spamming the checkbox and the request took a little longer?

Reply:
@AndrewJakubowicz I went back over this and tested it (clicking furiously! 😂), including with throttling, and the current closure behavior is the correct one. 

The output will say "Saving..." until it has returned from the request, and when it does finish, we are checking against the original state that the click was made during. Subsequent clicks do the same thing, eventually (not noticeable on a decent connection) settling on the final, correct state no matter how many times and how fast the checkbox is clicked.

That being said, rendering again 2 seconds afterwards would create this weird situation. I removed that second call to render, effectively leaving a "Saved!" text behind - which is better than trying to mess with state as it is in transit. That was a nice catch, thank you 👍 

For the other checkbox, there is less happening because it is not part of permissions and just a flag on the user. I'll leave that as it is, and only include this improvement for the school admin permission flag.